PTV-1887 - linting, formatting, updating reps, fixing tests

eapearson opened the pull request:
This set of changes aims to put the codebase into a stable, updated state in order to be able to make further changes.
The heart of, and initial set of, the changes was to apply uniform formatting via black, linting via pylint and sonarlint. 
Once these fixes were made, tests were run and any broken tests fixed until there were no errors.
However, once tests were passing, it was clear that not all tests were being invoked - code coverage was at about 50% and there were many tests skipped due to incorrect async usage and a few other issues. In fact, code coverage reports were not being generated either.
So the next effort was to update all dependencies, including adding a missing dependency required for asynchronous testing. Once through this, coverage was up to 84%.
Finally, once the changes were captured in a pull request, a series of changes were made to address any issues raised by sonar cloud or codify until they were all resolved. Some of these issues are specific to repeated values used in tests, which were simply annotated with # NOSONAR rather than rework the tests, as there are dozens if not hundreds of inline test data, and probably a 30-40 instances of duplicated test data which needed annotating.
There are no functional changes made around design. All code changes were made to address code quality issues raised by linting and related tools.

I did not include some changes that would help keep the codebase in shape. E.g. there should be tools (preferably docker-based) to apply formatting, linting checks, certainly ad-hoc but also via pre-commit hook and in the GitHub actions. Other welcome additions would be tightening up the type annotations, including mypy (currently 69 errors, and would require several more dependencies), and separating developer dependencies from those required by the service (runtime.)

eapearson committed:
ignore vscode settings [PTV-1887]

eapearson committed:
format  [PTV-1887]

eapearson committed:
sort dependencies  [PTV-1887]

eapearson committed:
actually vscode config was already checked in, so update  [PTV-1887]

eapearson committed:
many formatting and linting fixes  [PTV-1887]
still more to go, as there are some more serious issues like access to private members, and out of date dependencies, but this puts us on a good footing for any future change w/less format spam.

eapearson committed:
deprecated  [PTV-1887]

eapearson committed:
fix tests [PTV-1887]

eapearson committed:
add coverage reports, allow test path parameter [PTV-1887]

eapearson committed:
update dependencies, fix skipped tests, fix broken tests and code, use Python 3.11.4 [PTV-1887]

eapearson committed:
disable some sonarlint warnings [PTV-1887]

eapearson committed:
remove unused import [PTV-1887]

eapearson committed:
address couple more sonarcloud issues [PTV-1887]

eapearson committed:
last sonarcloud changes? [PTV-1887]

eapearson commented:
Hello @bio-boris. Here are the changes without the functional changes to fix and improve uploading. I'll start working on those next,

eapearson committed:
add auto asyncio mode for tests, remove annotations [PTV-1887]

eapearson committed:
replace references to Python 3.9.x w/ 3.11.4, repair README.md markdown a bit too [PTV-1887]

eapearson committed:
add release note [PTV-1887]

eapearson committed:
ignore PyCharm .idea directory [PTV-1887]

eapearson committed:
run black again [PTV-1887]

eapearson committed:
some markdown fixing and reformatting [PTV-1887]
 Not perfect. The readme has dozens of issues reported.

eapearson committed:
remove unused imports [PTV-1887]

eapearson committed:
add black and pymarkdownlint for development time [PTV-1887]

eapearson committed:
more markdown fixes, yuck [PTV-1887]

eapearson committed:
remove unused variables [PTV-1887]

eapearson committed:
use lower_snake_case for function name [PTV-1887]

eapearson committed:
one more silly set of markdown formatting fixes [PTV-1887]
just to satisfy remark-lint used by codacy, as it uses rules not supported by PyCharm. So annoying.

eapearson commented:
@bio-boris I've made a few changes since yesterday. Mostly just around release notes, and a few odds and ends.
It would be nice to get control over codacy and other code quality tools to get them to align with developer tools. It mostly does, but e.g. codacy uses "remark-lint" whereas markdownlint seems more popular and widely supported.

After this, I'll have a PR with the functional changes for fixing upload.

eapearson committed:
another silly set of markdown formatting fixes [PTV-1887]

bio-boris commented on import_specifications/generate_import_template.py:
```diff
@@ -20,45 +20,64 @@
 _NMS_URLS = {
```
LGTM

bio-boris commented on requirements.txt:
```diff
@@ -1,17 +1,20 @@
-python-magic==0.4.24
\ No newline at end of file
+frozendict==2.3.8
+globus-sdk==3.23.0
```
hopefully this globus update still 
works

bio-boris commented on run_tests.sh:
```diff
@@ -2,4 +2,13 @@
 DIR="$( cd "$( dirname "$0" )" && pwd )"
```
LGTM

bio-boris commented on scripts/prune_acls.py:
```diff
@@ -117,13 +127,13 @@ def get_old_acls():
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```
LGTM

bio-boris commented on scripts/refresh_token.py:
```diff
@@ -36,52 +36,54 @@
-#with open('/root/.globus.cfg', 'wb') as configfile:
-with open('/root/.globus.cfg', 'w') as configfile:
+# with open('/root/.globus.cfg', 'wb') as configfile:
+with open("/root/.globus.cfg", "w") as configfile:
```
LGTM

bio-boris commented on staging_service/AutoDetectUtils.py:
```diff
@@ -1,15 +1,44 @@
+    _MAPPINGS: Union[None, dict[str, Any]] = None  # expects to be set by config
+
+    @classmethod
+    def set_mappings(cls, mappings_config: MappingsConfig):
```
I'm a bit confused about why these are being changed to class methods, as the rest of the codebase uses static methods.

bio-boris commented on staging_service/JGIMetadata.py:
```diff
@@ -1,22 +1,20 @@
-from .utils import Path
+import os
```
LGTM

bio-boris commented on staging_service/__main__.py:
```diff
@@ -1,9 +1,11 @@
-from .app import app_factory
-from aiohttp import web
 import asyncio
```
LGTM

bio-boris commented on staging_service/app_error_formatter.py:
```diff
@@ -4,7 +4,7 @@
 from pathlib import Path
 
-from .import_specifications.file_parser import ErrorType, Error
+from .import_specifications.file_parser import Error, ErrorType
```
LGTM

bio-boris commented on staging_service/auth2Client.py:
```diff
@@ -4,13 +4,14 @@
 @author: gaprice@lbl.gov
 modified for python3 and authV2
```
LGTM

bio-boris commented on staging_service/autodetect/GenerateMappings.py:
```diff
@@ -20,85 +21,111 @@
 from collections import defaultdict
 
-from staging_service.autodetect.Mappings import *
+from staging_service.autodetect import Mappings
```
LGTM

bio-boris commented on staging_service/autodetect/GenerateMappings.py:
```diff
@@ -20,85 +21,111 @@
-    phenotype_set_id: "Phenotype Set",
-    escher_map_id: "EscherMap",
-    import_specification: "Import Specification",
+    Mappings.sra_reads_id: "SRA Reads",
```
Sad that these are long now :(

bio-boris replied:
These really make the readability much worse.

bio-boris commented on staging_service/autodetect/Mappings.py:
```diff
@@ -31,7 +31,8 @@
 MEDIA = "KBaseBiochem.Media"
 
-# These ID mappings should be available in dropdown_order in staging_upload.json in the narrative service
+# These ID mappings should be available in dropdown_order in staging_upload.json
```
LGTM

bio-boris commented on staging_service/globus.py:
```diff
@@ -1,8 +1,10 @@
-import aiohttp
-import aiofiles
-import os
 import configparser
```
Seems fine

bio-boris commented on staging_service/import_specifications/file_parser.py:
```diff
@@ -6,12 +6,10 @@
 from collections.abc import Callable
```
LGTM

bio-boris commented on staging_service/import_specifications/file_writers.py:
```diff
@@ -220,11 +248,14 @@ def _write_excel_row(sheet: Worksheet, row: int, contents: list[Any]):
-        sheet.column_dimensions[get_column_letter(column_cells[0].column)].width = length
+        sheet.column_dimensions[
+            get_column_letter(column_cells[0].column)
+        ].width = length
```
lgtm

bio-boris commented on staging_service/import_specifications/individual_parsers.py:
```diff
@@ -31,10 +30,14 @@
+    f"{_DATA_TYPE} (\\w+){_HEADER_SEP} "
+    + f"{_COLUMN_STR} (\\d+){_HEADER_SEP} {_VERSION_STR} (\\d+)"
+)
 
```
LGTM

bio-boris commented on staging_service/metadata.py:
```diff
@@ -28,18 +30,18 @@ async def stat_data(path: Path) -> dict:
 def _file_read_from_tail(file_path):
     upper_bound = min(1024, os.stat(file_path).st_size)
-    with open(file_path, "r") as file:
+    with open(file_path, "r", encoding="utf-8") as file:
```
Hopefully the platform encoding  was UTF-8!

bio-boris commented on staging_service/metadata.py:
```diff
@@ -75,7 +77,7 @@ async def add_upa(path: Path, UPA: str):
             data = decoder.decode(data)
     else:
-        data = await _generate_metadata(path)  # TODO performance optimization
+        data = await _generate_metadata(path, None)
```
How did this ever work if _generate_metadata expected 2 inputs?

bio-boris commented on staging_service/utils.py:
```diff
@@ -187,7 +189,7 @@ def _add_acl(self, user_identity_id: str, shared_directory_basename: str):
             logging.info(response)
             logging.info(
-                "Shared %s with %s\n" % (shared_directory_basename, user_identity_id)
+                "Shared %s with %s\n", shared_directory_basename, user_identity_id
```
could use fstrings here as well

eapearson committed:
import all Mappings symbols explicitly [PTV-1887]

eapearson committed:
remove pymarkdownlnt (not used), revert globus-sdk to 1.6.1  [PTV-1887]
There are breaking changes; it should be updated, but probably not in this set of changes.

bio-boris commented on requirements.txt:
```diff
@@ -1,17 +1,19 @@
-python-magic==0.4.24
\ No newline at end of file
+frozendict==2.3.8
+globus-sdk==1.6.1
```
LGTM

bio-boris commented on tests/import_specifications/test_file_parser.py:
```diff
@@ -242,25 +274,28 @@ def test_ParseResults_init_w_error_success():
-    err = 'Exectly one of results or errors must be supplied'
-    parseResults_init_fail(None, None, ValueError(err))
-    parseResults_init_fail(PR_RESULTS, PR_ERROR, ValueError(err))
+def test_parse_results_init_fail():
```
LGTM

bio-boris commented on tests/import_specifications/test_file_parser.py:
```diff
@@ -406,50 +468,72 @@ def test_parse_import_specification_multiple_specs_and_parser_error():
-    parser3.assert_called_once_with(Path("x.tsv"))
-    logger.assert_not_called()
\ No newline at end of file
+    parser3.assert_called_once_with(Path("x.tsv"))  # NOSONAR python:S1192
```
I think we can ignore sonarcloud, not a big worry

bio-boris commented on tests/import_specifications/test_file_writers.py:
```diff
@@ -193,84 +191,140 @@ def test_file_writers_fail():
 
 
-def file_writers_fail(path: Path, types: dict, expected: Exception):
+def file_writers_fail(path: Union[Path, None], types: dict, expected: Exception):
```
LGTM

bio-boris commented on tests/pyproject.toml:
```diff
@@ -3,4 +3,5 @@ minversion = "6.0"
-]
\ No newline at end of file
+]
+asyncio_mode = "auto"
```
This is some sort of test fixture

bio-boris commented on tests/import_specifications/test_individual_parsers.py:
```diff
@@ -244,22 +368,34 @@ def test_xsv_parse_fail_empty_file(temp_dir: Path):
-    _xsv_parse_fail(temp_dir, ["Data type: foo; Columns: 3; Version: 1\n"], parse_csv, err)
+    _xsv_parse_fail(
+        temp_dir,
+        ["Data type: foo; Columns: 3; Version: 1\n"],  # NOSONAR python:S1192
```
OK

bio-boris commented on tests/test_app.py:
```diff
@@ -1578,33 +1682,39 @@ async def _write_bulk_specification_json_fail(json: Any, err: str):
     await _write_bulk_specification_json_fail(
         {"output_directory": "foo", "output_file_type": "CSV", "types": {"a": "fake"}},
-         "The value for data type a must be a mapping"
+        "The value for data type a must be a mapping",
```
ok

bio-boris commented on tests/test_app_error_formatter.py:
```diff
@@ -91,21 +96,24 @@ def test_format_import_spec_errors_all_the_errors_with_tabs():
+            "tab": "tab2",
+        },
+        {
+            "type": "multiple_specifications_for_data_type",
```
ok

bio-boris commented on tests/test_autodetect.py:
```diff
@@ -176,49 +214,59 @@ def test_zip_mappings():
+                "suffix": "Gz",
+                "file_ext_type": ["CompressedFileFormatArchive"],
+            },
+            {"prefix": "some.dots", "suffix": "gff3.gz", "file_ext_type": ["GFF"]},
```
lgtm

bio-boris commented on tests/test_metadata.py:
```diff
@@ -73,8 +66,9 @@ async def _incomplete_metadata_file_update(temp_dir, metadict, source):
         "path": "user_path",
-        "size": 1280
-        }
+        "size": 1280,
```
ok

bio-boris commented on tests/test_utils.py:
```diff
@@ -34,21 +34,23 @@ def bootstrap_config():
+
 def check_excel_contents(
-    wb: openpyxl.Workbook,
+    workbook: openpyxl.Workbook,
```
ok

bio-boris reviewed (approved):
LGTM
